exchanges/okx/okx_websocket_requests.go

```diff
@@ -0,0 +1,218 @@
+	}
+	if arg.NewQuantity <= 0 && arg.NewPrice <= 0 {
+		return nil, errInvalidNewSizeOrPriceInformation
+	}
+
+	id := strconv.FormatInt(ok.Websocket.AuthConn.GenerateMessageID(false), 10)
+
+	var orderResp []OrderData
+	err := ok.SendAuthenticatedWebsocketRequest(ctx, id, "amend-order", []AmendOrderRequestParams{*arg}, &orderResp)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(orderResp) != 1 {
+		return nil, errNoValidResponseFromServer
+	}
+
+	if orderResp[0].SCode > 0 {
+		return nil, fmt.Errorf("error code: %d message: %s", orderResp[0].SCode, orderResp[0].SMessage)
+	}
+
+	return &orderResp[0], nil
+}
+
+// WsAmendMultipleOrders a request through the websocket connection to amend multiple trade orders.
+func (ok *Okx) WsAmendMultipleOrders(ctx context.Context, args []AmendOrderRequestParams) ([]OrderData, error) {
+	if len(args) == 0 {
+		return nil, fmt.Errorf("%T: %w", args, common.ErrNilPointer)
+	}
+
+	for x := range args {
+		if args[x].InstrumentID == "" {
+			return nil, errMissingInstrumentID
+		}
+		if args[x].ClientOrderID == "" && args[x].OrderID == "" {
+			return nil, errMissingClientOrderIDOrOrderID
+		}
+		if args[x].NewQuantity <= 0 && args[x].NewPrice <= 0 {
+			return nil, errInvalidNewSizeOrPriceInformation
+		}
+	}
+
+	id := strconv.FormatInt(ok.Websocket.AuthConn.GenerateMessageID(false), 10)
+
+	// Opted to not check scode here as some orders may be successful and some
+	// may not. So return everything to the caller to handle.
+	var orderResp []OrderData
+	return orderResp, ok.SendAuthenticatedWebsocketRequest(ctx, id, "batch-amend-orders", args, &orderResp)
+}
+
+// SendAuthenticatedWebsocketRequest sends a websocket request to the server
+// TODO: Websocket request endpoint rate limits are shared with their REST API counterparts. A future update should
+// access request level rate limits and update the rate limiter accordingly.
+func (ok *Okx) SendAuthenticatedWebsocketRequest(ctx context.Context, id, operation string, payload, response any) error {
+	if id == "" || operation == "" || payload == nil {
+		return errInvalidWebsocketRequest
+	}
+
+	outbound := &struct {
+		ID        string `json:"id"`
+		Operation string `json:"op"`
+		Arguments any    `json:"args"`
+	}{
+		ID:        id,
+		Operation: operation,
+		Arguments: payload,
+	}
+
+	incoming, err := ok.Websocket.AuthConn.SendMessageReturnResponse(ctx, request.Unset, id, outbound)
+	if err != nil {
+		return err
+	}
+
+	intermediary := struct {
+		ID        string `json:"id"`
+		Operation string `json:"op"`
+		Code      int    `json:"code,string"`
+		Message   string `json:"msg"`
+		Data      any    `json:"data"` // pointer to type
+		InTime    string `json:"inTime"`
+		OutTime   string `json:"outTime"`
+	}{
+		Data: response,
+	}
+
+	if err := json.Unmarshal(incoming, &intermediary); err != nil {
+		return err
+	}
+
+	if intermediary.Code > 1 {
+		return fmt.Errorf("error code: %d message: %s", intermediary.Code, intermediary.Message)
+	}
```

Comment:
Order code 2 was previously handled. Your comments on partial success in those functions likely won't get hit with a response of 2

Reply:
Yeah good catch; I bumped it to 2 so that anything beyond that can fail here, 1&2 codes don't actually need to fail here. None of my testing actually returned a code 2 that's why I must have forgotten to put it in. Example:

```
Okx wss://ws.okx.com:8443/ws/v5/private: Sending message: {"id":"1","op":"batch-orders","args":[{"instId":"BTC-USDT","tdMode":"cash","ccy":"USDT","side":"buy","ordType":"limit","sz":"0.0001","px":"-1"},{"instId":"BTC-USDT","tdMode":"cash","ccy":"USDT","side":"buy","ordType":"limit","sz":"0.0001","px":"20000"}]}
Okx wss://ws.okx.com:8443/ws/v5/private: Received response [1/1]: {"id":"1","op":"batch-orders","code":"2","msg":"","data":[{"tag":"","ts":"1728858915989","ordId":"","clOrdId":"","sCode":"51000","sMsg":"Parameter px  error"},{"tag":"","ts":"1728858915990","ordId":"1891010883570696192","clOrdId":"","sCode":"0","sMsg":"Order successfully placed."}],"inTime":"1728858915989134","outTime":"1728858915990899"}
```

Correct code 1 is a failure but the data holds the specifics of the error in sCode and sMsg and that can be checked in the calling function.

Reply:
I get that you're trying to put the onus on the caller, but you also don't give the caller the opportunity to do so since you only return the `Data` property and nothing else. Then with this:
```go
if len(orderResp) != 1 {
		return nil, errNoValidResponseFromServer
	}
```
There could be a `Message` and `Code` from the parent which could contain details but its instead thrown out and you don't know why.

Reply:
All code 1 and code 2 API examples don't produce a message; 1 is a total failure and 2 is a partial failure which have data returned with specific scode and smsg to each action. From testing anything above 2 which has a message component; I have only seen request errors pertaining to operation string, connection authentication issues and they don't send any data back for the caller. The only reason I added the length check was to protect against a possible panic. 

Happy to add in whatever you want though send me a diff on what you propose and the response you have found anything while testing and I will update it.